ci-operator/step-registry/cucushift/installer/check/gcp/private/cucushift-installer-check-gcp-private-commands.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+GCP_REGION="${LEASED_RESOURCE}"
+
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ret=0
+
+echo "INFO: (1/4) Checking publish policy..."
+publish_policy=$(oc get configmap -n kube-system cluster-config-v1 -oyaml | grep 'publish:')
+echo "INFO: ${publish_policy}"
+if [[ "${publish_policy}" =~ "publish: Internal" ]]; then
+    echo "INFO: Publish policy check passed."
+else
+    echo "ERROR: Publish policy check failed."
+    ret=$(( $ret | 1 ))
```

Comment:
Why not directly  `+ 1` instead of `| 1`, any check failure will lead ret to be a non-zero value.

Reply:
It's for detecting which check(s) failed by the exit code, for example, if only "Publish policy check failed", the exit code would 1, and if both "Base domain record-sets check failed" and "External address check failed" the exit code would be 6.